### DISCUSSION TITLE
macOS: Ghostty using 100% CPU

### DISCUSSION THREAD

**Discussion by stuartcarnie** [ghostty-org/ghostty#3723] [Feature Requests, Ideas] [answered]
Noticed that Ghostty was using 100% CPU, so I grabbed a CPU profile under Instruments and found that the hottest call stack was related to the Quick Terminal feature:

![img](https://cdn.discordapp.com/attachments/1322555981103632447/1322560293984800789/CleanShot_2024-12-28_at_06.41.142x.png?ex=677151c2&is=67700042&hm=44d828e85fe45540883b199ea6eb5aa50c8b57efc89d9871f6a158982b63ce4c&)

When I drill into that stack, `QuickTerminalController.syncAppearance` is a hot function. I currently don't have the window open, and I believe the culprit is this code:

https://github.com/ghostty-org/ghostty/blob/196af6f2ca7221b38ee14c8f286cc99e0a7108fa/macos/Sources/Features/QuickTerminal/QuickTerminalController.swift#L310-L318

Given the window isn't open, this will repeatedly call the `syncAppearance` function. 

> [!NOTE]
>
> Re-opening the Quick Terminal reduces the CPU to nominal levels again.

To re-trigger the high CPU usage, force a reload of your configuration and watch it spike again.

**Comment by habahabahaba** [comment]
I also had Ghostty taking close to 100% CPU, while being idle, on my M2 Air Sonoma 14.7; but as it didn't happen again, I can't say if it was Quick-Terminal related.

**Comment by mitchellh** [comment, marked as answer]
Thank you for finding this! This was the source of CPU issues on the regular terminal long ago, I forgot to apply similar fixes to the quick terminal. Approved: #3998

**Reply by franklouwers** [reply]
Am afraid this doesn't fix it. I still see the same behavior in 1.1.2 (but can't validate it's the quickterm yet)
![CleanShot 2025-03-22 at 23 06 31@2x](https://github.com/user-attachments/assets/70aebbc0-cef6-4652-b20d-8ee65cfdc78f)

**Reply by taylrfnt** [reply]
In the future, please open a new post and simply mention the solved/closed discussion.  It makes triaging issues much more difficult when folks add new reports onto solved posts, especially ones that are months old

**Reply by franklouwers** [reply]
ok thanks
